### ISSUE TITLE
only show stack trace in dev mode

### ISSUE THREAD

**Issue by williballenthin** [mandiant/capa#1515]
ensure that we only show a stack trace when users might want to (such as developers). otherwise provide a last line of defense exception handler to suggest what users can do. 

https://www.bitecode.dev/p/why-and-how-to-hide-the-python-stack

**Comment by aaronatp** [comment]
Hi @williballenthin I'd like to contribute to this issue. Thank you for linking this article! I have read it and it is very helpful.

As the article suggests, a callback could show the stack trace if `sys.flags.dev_mode` is True or print another message if it is false. Do you think something like the following would work here? I was also thinking about linking the capa Issues page and asking users to report any unexpected exceptions - is this potentially a good idea?

```
import sys

SHOW_STACK_TRACE = sys.flags.dev_mode

def last_resort_exception_handler(exctype, value, traceback):   
    if SHOW_STACK_TRACE:
        sys.__excepthook__(exctype, value, traceback)
    else:
        print(f"Unexpected exception raised: {exctype}. Please run capa in Python development mode "
            to see the stack trace (https://docs.python.org/3/library/devmode.html#devmode)")

sys.excepthook = last_resort_exception_handler
```

**Comment by williballenthin** [comment]
I think your idea is generally great. Lets try this out and see how it improves user experience.

One thought: lets use the `--debug` flag passed via CLI versus `dev_mode`. That's because "It should not be more verbose than the default if the code is correct". https://github.com/mandiant/capa/blob/647abb669f29c03d4c222ccf30a15d094c627d2e/capa/main.py#L877

Please go ahead and open a PR and we can review it together.

**Comment by aaronatp** [comment]
@williballenthin awesome thank you! What you are saying about the `--debug` flag makes sense. I'll make that adjustment and open a PR!

**Comment by mr-tz** [comment]
closed via #1860
